spec/lib/bitmap_editor_spec.rb

```diff
@@ -0,0 +1,107 @@
+require 'bitmap_editor'
+
+RSpec.describe BitmapEditor do
+  context '.run' do
+    context 'with wrong file path' do
+      it 'returns an error message asking for a correct path' do
+        expect do
+          BitmapEditor.new.run('wrong/path/file.txt')
+        end.to output("Please provide correct file\n").to_stdout
+      end
```

Comment:
There's quite a lot of repetition and boilerplate in these tests that you can clean up with a bit of refactoring.
Because we're running the same command each time, we can make that the subject:
```
subject:(run) { described_class.new.run(path) }
```
Then each test can be like
```
context 'with wrong file path' do
  let(:path) { 'wrong/path/file.txt' }

  it 'returns an error message asking for a correct path' do
    expect(run).to output("Please provide correct file\n").to_stdout
  end
end
```
Having done that, most complexity is now in the expected result, so we can move that too
```
context 'with wrong file path' do
  let(:path) { 'wrong/path/file.txt' }
  let(:expected_text) { "Please provide correct file\n" }

  it 'returns an error message asking for a correct path' do
    expect(run).to output(expected_text).to_stdout
  end
end
```
At that point, you should have a load of tests where just the `let`s change, so at that point you can move all the expectations into a single `shared_example`:
```
subject:(run) { described_class.new.run(path) }

shared_examples "a valid stdout response" do
  it 'returns the correct message given the path' do
    expect(run).to output(expected_text).to_stdout
  end
end

context 'with wrong file path' do
  let(:path) { 'wrong/path/file.txt' }
  let(:expected_text) { "Please provide correct file\n" }
  
  it_behaves_like 'a valid stdout response'
end
```